ui/app/models/job.js

```diff
@@ -366,4 +366,17 @@ export default class Job extends Model {
       return this.variables?.findBy('path', `nomad/jobs/${this.plainId}`);
     }
   }
+
+  // TODO: This async fetcher seems like a better fit for most of our use-cases than the above getter (which cannot do async/await)
+  async getPathLinkedVariable() {
+    await this.variables;
+    if (this.parent.get('id')) {
+      return await this.variables?.findBy(
+        'path',
+        `nomad/jobs/${JSON.parse(this.parent.get('id'))[0]}`
+      );
+    } else {
+      return await this.variables?.findBy('path', `nomad/jobs/${this.plainId}`);
+    }
+  }
```

Comment:
This is, IMO, the most notable change I'm making in this PR. See the `pathLinkedVariable()` getter above this for how we've traditionally done it.

For the past few months, looking at a variable in the Web UI has been in a context where we'd have just gotten variables loaded. The operation, though, isn't immediate or promise-friendly. Frankly if this gets merged I intend to do away with `pathLinkedVariable()` in favour of `getPathLinkedVariable()` and make the whole operation more awaitable.

This lets us wait for the value of the job's (or group's, or task's, as you'll see in a moment) related variable without making any assumptions about what's already been loaded (see for example `_fetchParentJob` in the task model)